File: golang/arithmetic_slices.go
package golang

func numberOfArithmeticSlices(A []int) int {
	ans := 0
	var temp []int
	n := len(A)

	if n < 3 {
		return 0
	}

	for i := 1; i < n; i++ {
		temp = append(temp, A[i]-A[i-1])
	}

	count := 0
	for j := 1; j < len(temp); j++ {
		if temp[j] == temp[j-1] {
			count++
		} else {
			count = 0
		}
		ans += count
	}

	return ans
}

File: python3/arithmetic_slices.py
# Space   : O(n)
# Time    : O(n)

class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        ans = 0
        temp = []
        n = len(A)

        if n < 3:
            return 0

        for i in range(1, n):
            temp.append(A[i] - A[i-1])

        count = 0
        for j in range(1, len(temp)):
            if temp[j] == temp[j-1]:
                count += 1
            else:
                count = 0
            ans += count

        return ans
